lib/ansible/modules/database/mysql/mysql_variables.py

```diff
@@ -177,27 +222,52 @@ def main():
         else:
             module.fail_json(msg="unable to find %s. Exception message: %s" % (config_file, to_native(e)))
 
+    mysqlvar_val = None
+    var_in_mysqld_auto_cnf = None
+
     mysqlvar_val = getvariable(cursor, mysqlvar)
     if mysqlvar_val is None:
         module.fail_json(msg="Variable not available \"%s\"" % mysqlvar, changed=False)
+
     if value is None:
         module.exit_json(msg=mysqlvar_val)
-    else:
-        # Type values before using them
-        value_wanted = typedvalue(value)
-        value_actual = typedvalue(mysqlvar_val)
-        if value_wanted == value_actual:
-            module.exit_json(msg="Variable already set to requested value", changed=False)
-        try:
-            result = setvariable(cursor, mysqlvar, value_wanted)
-        except SQLParseError as e:
-            result = to_native(e)
-
-        if result is True:
-            module.exit_json(msg="Variable change succeeded prev_value=%s" % value_actual,
-                             changed=True, queries=executed_queries)
+
+    if mode in ('persist', 'persist_only'):
+        var_in_mysqld_auto_cnf = check_mysqld_auto(module, cursor, mysqlvar)
+        if var_in_mysqld_auto_cnf is None:
```

Comment:
I think you should only do this whole `if` if `mode == 'persist_only'`.

Reply:
Probably i understood incorrect, why?
We need to get value from mysqld-auto.cnf ```var_in_mysqld_auto_cnf = check_mysqld_auto(module, cursor, mysqlvar)``` and check it for ```persist``` mode too

Reply:
If `mode == 'persist'`, you want that *both* the global value currently set, and the value stored in the config file, are both equal to `value`. If you read both values, but overwrite one with the other, you cannot compare both to `value`.

Reply:
it seems work, PTAL :)